x/ecocredit/spec/07_client.md

```diff
@@ -1735,6 +1933,177 @@ Example Output:
 }
 ```
 
+### BalancesByBatch
+
+The `BalancesByBatch` endpoint allows users to query all credit balances from a
+given credit batch.
+
+```bash
+/regen/ecocredit/v1/balances-by-batch/{batch_denom}
+/regen/ecocredit/v1/batches/{batch_denom}/balances
+```
+
+Example:
+
+```bash
+curl localhost:1317/regen/ecocredit/v1/batches/C02-001-20210101-20250101-001/balances
+```
+
+Example Output:
+
+```bash
+{
+  "balances": [
+    {
+      "address": "regen1qwa9xy0997j5mrc4dxn7jrcvvkpm3uwuldkrmg",
+      "batch_denom": "C02-001-20210101-20250101-001",
+      "tradable_amount": "92.0",
+      "retired_amount": "97",
+      "escrowed_amount": "30.0"
+    }
+  ],
+  "pagination": {
+    "next_key": "BwEJABQDulMR5S+lTY8VaafpDwxlg7jx3AAF"
+  }
+}
+```
+
+### AllBalances
+
+The `AllBalances` endpoint allows users to query all credit balances.
+
+```bash
+/regen/ecocredit/v1/all-balances
+/regen/ecocredit/v1/balances
+```
+
+Example:
+
+```bash
+curl localhost:1317/regen/ecocredit/v1/balances
+```
+
+Example Output:
+
+```bash
+{
+  "balances": [
+    {
+      "address": "regen1qwa9xy0997j5mrc4dxn7jrcvvkpm3uwuldkrmg",
+      "batch_denom": "C02-001-20210101-20250101-001",
+      "tradable_amount": "92.0",
+      "retired_amount": "97",
+      "escrowed_amount": "30.0"
+    }
+  ],
+  "pagination": {
+    "next_key": "BwEJABQDulMR5S+lTY8VaafpDwxlg7jx3AAF"
+  }
+}
+```
+
+### ClassCreatorAllowlist
+
+The `ClassCreatorAllowlist` endpoint allows users to query the credit class
+creator allowlist enable setting.
+
+```bash
+/regen/ecocredit/v1/class-creator-allowlist
+```
+
+Example:
+
+```bash
+curl localhost:1317/regen/ecocredit/v1/class-creator-allowlist
+```
+
+Example Output:
+
+```bash
+{
+  "enabled":"true"
+}
+```
+
+### AllowedClassCreators
+
+The `AllowedClassCreators` endpoint allows users to query allowed credit class
+creators list.
+
+```bash
+/regen/ecocredit/v1/allowed-class-creators
+```
+
+Example:
+
+```bash
+curl localhost:1317/regen/ecocredit/v1/allowed-class-creators
+```
+
+Example Output:
+
+```bash
+{
+  "class_creators":[
```

Comment:
noticing that some of these are snake_case and others are camelCase - are some of these outdated?

Reply:
rpc returns camel case and rest returns snake case

Reply:
```
grpcurl -plaintext -d '{"batch_denom": "C01-001-20200101-20210101-001"}' localhost:9090 regen.ecocredit.v1.Query/BalancesByBatch

...

{
  "balances": [
    {
      "address": "regen1xm4ty25m95gvv6h7exr65luh6uv3fqyypnmgmw",
      "batchDenom": "C01-001-20200101-20210101-001",
      "tradableAmount": "1000",
      "retiredAmount": "500",
      "escrowedAmount": "0"
    }
  ],
  "pagination": {
    "total": "1"
  }
}
```

```
curl localhost:1317/regen/ecocredit/v1/batches/C01-001-20200101-20210101-001/balances

...

{
  "balances": [
    {
      "address": "regen1xm4ty25m95gvv6h7exr65luh6uv3fqyypnmgmw",
      "batch_denom": "C01-001-20200101-20210101-001",
      "tradable_amount": "1000",
      "retired_amount": "500",
      "escrowed_amount": "0"
    }
  ],
  "pagination": {
    "next_key": null,
    "total": "1"
  }
}
```